### PR TITLE
Add IsItChristmas IA

### PR DESCRIPTION
**Pull request by stevenmg**
# IsItChristmas

**What does your Instant Answer do?**
It adds a simple and fun answer to the question of whether or not it is currently Christmas.

**What problem does your Instant Answer solve (Why is it better than organic links)?**
It allows people to find an answer to the important question of whether or not it is Christmas without the need to click through to isitchristmas.com.

**What is the data source for your Instant Answer? (Provide a link if possible)**
The current Date combined with my knowledge of the fact that Christmas is always December 25th

**Why did you choose this data source?**
This question seems invalid in the current context.

**Are there any other alternative (better) data sources?**
Nope.

**What are some example queries that trigger this Instant Answer?**
"is it christmas"

**Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)**
Anyone who loves Christmas

**Is this Instant Answer connected to a DuckDuckHack [Instant Answer idea](https://duck.co/ideas)?**
Nope

**Which existing Instant Answers will this one supercede/overlap with?**
None

**Are you having any problems? Do you need our help with anything?**
Using Codio it is impossible to change the date in order to ensure that the IA really will say "Yes" on December 25th so if someone could test that more thoroughly that would be great.

**What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)**
Is It Christmas?

## Checklist
Please place an 'X' where appropriate.

```
- [x] Added metadata and attribution information (https://duck.co/duckduckhack/metadata)
- [x] Wrote test file and added to t/ directory (https://duck.co/duckduckhack/goodie_tests)
- [x] Verified that instant answer adheres to design guidelines (https://duck.co/duckduckhack/design_styleguide)
- [] Verified that instant answer adheres to code styleguide (https://duck.co/duckduckhack/code_styleguide)
- [] Tested cross-browser compatibility

    Please let us know which browsers/devices you've tested on:
    - Windows 8
        - [] Google Chrome
        - [x] Firefox
        - [] Opera
        - [] IE 10

    - Windows 7
        - [] Google Chrome
        - [] Firefox
        - [] Opera
        - [] IE 8
        - [] IE 9
        - [] IE 10

    - Windows XP
        - [] IE 7
        - [] IE 8

    - Mac OSX
        - [] Google Chrome
        - [] Firefox
        - [] Opera
        - [] Safari

    - iOS (iPhone)
        - [] Safari Mobile
        - [] Google Chrome
        - [] Opera

    - iOS (iPad)
        - [] Safari Mobile
        - [] Google Chrome
        - [] Opera

    - Android
        - [] Firefox
        - [] Native Browser
        - [] Google Chrome
        - [] Opera

```

### CODE REVIEW

**Comment by moollaza** [comment]
@stevenmg thanks for contributing! We really appreciate it :smiley: 

We'll be sure to give you some feedback shortly!

File: lib/DDG/Goodie/IsItChristmas.pm [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+# Triggers
+triggers any => "isitchristmas", "is it christmas";
+
+# Handle statement
+handle remainder => sub {
+    my ($y, $m, $d) = Date::Calc::Today();
+    print($m);
+    print($d);
+
+    if ($m == 12 && $d == 25) {
```
Comment (puskin94) [inline lib/DDG/Goodie/IsItChristmas.pm]: I think that `($m == 12 && $d == 25) ? return "Yes" : return "No";` is styled better ;)

File: lib/DDG/Goodie/IsItChristmas.pm [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+            web     => ['http://www.steveglick.net/', 'Steve Glick'],
+            twitter => "stevenmglick";
+
+# Triggers
+triggers any => "isitchristmas", "is it christmas";
+
+# Handle statement
+handle remainder => sub {
+    my ($y, $m, $d) = Date::Calc::Today();
+    print($m);
```
Comment (puskin94) [inline lib/DDG/Goodie/IsItChristmas.pm]: Is this a debug remnant?

File: lib/DDG/Goodie/IsItChristmas.pm (hunk 1 above)
Comment (mintsoft) [inline lib/DDG/Goodie/IsItChristmas.pm]: I would disagree! I'd much prefer
```perl
return "Yes" if ($m == 12 && $d == 25);
return "No";
```
or
```perl
return "No" unless ($m == 12 && $d == 25);
return "Yes";
```

File: lib/DDG/Goodie/IsItChristmas.pm [hunk 3]
```diff
@@ -0,0 +1,36 @@
+package DDG::Goodie::IsItChristmas;
+
```
Comment (mintsoft) [inline lib/DDG/Goodie/IsItChristmas.pm]: Can you add an abstract here? An example is: https://github.com/duckduckgo/zeroclickinfo-goodies/blob/master/lib/DDG/Goodie/ABC.pm#L2

File: lib/DDG/Goodie/IsItChristmas.pm [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+use DDG::Goodie;
+use Date::Calc;
+
+zci answer_type => "is_it_christmas";
+zci is_cached   => 1;
+
+# Metadata.  See https://duck.co/duckduckhack/metadata for help in filling out this section.
```
Comment (mintsoft) [inline lib/DDG/Goodie/IsItChristmas.pm]: You can delete this comment

File: t/IsItChristmas.t [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+use strict;
+use warnings;
+use Test::More;
+use DDG::Test::Goodie;
+
+zci answer_type => "is_it_christmas";
+zci is_cached   => 1;
+
+ddg_goodie_test(
```
Comment (mintsoft) [inline t/IsItChristmas.t]: I think using MockTime would be pretty important here, so we can guarantee the actual functionality. It's not very clean, however there is an example here: https://github.com/duckduckgo/zeroclickinfo-goodies/blob/master/t/CalendarToday.t#L87

**Comment by puskin94** [comment]
@mintsoft only as a comparison... Why do you prefer your `if` solution instead of mine?

**Comment by mintsoft** [comment]
@puskin94 A few relatively minor reasons; it's more certainly more perlistic. The main advantage is that I think it's clearer to see the default vs the exception case whereas with a ternary it's nowhere near as obvious imo

**Comment by puskin94** [comment]
@mintsoft Yep, it is more and more explicit. But, in my opinion and in the style guide in my own mind, I think that compact is better ;)

**Comment by mintsoft** [comment]
@puskin94 I think it's probably a matter of style mainly. In my experience I've found that clarity is the most important thing, especially when working in a team; however I'm sure there are other philosophies.

**Comment by stevenmg** [comment]
Thanks for the input guys. I typically prefer more clear over more compact. Does the style guide specify which one should be used?

Also, I like the idea of using mock time for tests, I will work on that later.

**Comment by mintsoft** [comment]
@stevenmg it's totally up to you; whichever you prefer!

**Comment by mintsoft** [comment]
@stevenmg also btw I don't see any real problem with what you have currently there; there's certainly no need to change it unless you want to

**Comment by mintsoft** [comment]
@stevenmg I've added the Work in Progress label for now, whenever you're finished up just remove it :+1:

**Comment by stevenmg** [comment]
My mental style guide says to keep it as is (probably because I use Java a lot) so I think that is what I'll do unless there is another reason to change it.

**Comment by stevenmg** [comment]
I have added mock time for tests as well as 'is it xmas' as a trigger phrase. If anything else should be added let me know.

**Comment by stevenmg** [comment]
I was just looking at the Travis CI status for my earlier commits and it looks like the tests are failing for this IA. I am going to wait for the results of the latest commits and then see what I can do.

**Comment by stevenmg** [comment]
I just fixed up the test file. I think I'm done here.

File: lib/DDG/Goodie/IsItChristmas.pm [hunk 6]
```diff
@@ -0,0 +1,34 @@
+package DDG::Goodie::IsItChristmas;
+# ABSTRACT: Answers the question of whether or not it is Christmas with a simple yes or no answer
+
+use DDG::Goodie;
+use Date::Calc;
```
Comment (mintsoft) [inline lib/DDG/Goodie/IsItChristmas.pm]: Can you switch this to: `use DateTime;` as we already use this module in the Dates role

**Comment by javathunderman** [comment]
@mintsoft @stevenmg Is this ready to be merged?

**Comment by mintsoft** [comment]
@javathunderman almost, @abeyang does this look aright to you or would you prefer it to use the structured answer?

**Comment by javathunderman** [comment]
ping @mintsoft @abeyang

**Comment by moollaza** [comment]
Hey guys, we're discussing this IA internally with regards to whether or not it deserves to be an IA given the narowness of the queryspace. We may decide it needs to be generalized, or perhaps this should be handled by the coming Holidays IA.

I'll be sure to keep you updated!

**Comment by javathunderman** [comment]
Holidays IA? :)

**Comment by moollaza** [comment]
https://github.com/duckduckgo/zeroclickinfo-spice/pull/1701

**Comment by moollaza** [comment]
Hey guys, sorry for the long delay. I feel like the queries this IA handles should really be handled by the Holidays IA as it has a much broader scope.

We feel that this IA is a little *too* narrow in scope.

@stevenmg I'm really sorry we can't move forward with this but we really hope you're still interested in contributing! Is there anything else you'd like to work on? A [cheat sheet](https://duck.co/duckduckhack/goodie_cheat_sheets) perhaps?

**Pull request closed without merging**